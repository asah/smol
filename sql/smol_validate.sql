SET client_min_messages = warning;
CREATE EXTENSION IF NOT EXISTS smol;

-- Test smol_validate() function (line 2433)
-- This function is called by PostgreSQL during CREATE OPERATOR CLASS
-- It gets executed when the extension is created/loaded


-- Query all smol operator classes (validates they were created successfully)
SELECT opcname, opfmethod.amname
FROM pg_opclass opc
JOIN pg_am opfmethod ON opc.opcmethod = opfmethod.oid
WHERE opfmethod.amname = 'smol'
ORDER BY opcname;

-- Verify that smol indexes can be created with all supported types
DROP TABLE IF EXISTS t_validate CASCADE;
CREATE UNLOGGED TABLE t_validate (
    i2 int2,
    i4 int4,
    i8 int8,
    f4 float4,
    f8 float8,
    d date,
    t text
);

-- Create indexes to ensure operator classes are valid
CREATE INDEX idx_val_i2 ON t_validate USING smol(i2);
CREATE INDEX idx_val_i4 ON t_validate USING smol(i4);
CREATE INDEX idx_val_i8 ON t_validate USING smol(i8);
CREATE INDEX idx_val_f4 ON t_validate USING smol(f4);
CREATE INDEX idx_val_f8 ON t_validate USING smol(f8);
CREATE INDEX idx_val_d ON t_validate USING smol(d);
CREATE INDEX idx_val_t ON t_validate USING smol(t);

-- Verify all indexes were created successfully
SELECT indexname, indexdef
FROM pg_indexes
WHERE tablename = 't_validate'
ORDER BY indexname;

-- Cleanup
DROP TABLE t_validate CASCADE;
