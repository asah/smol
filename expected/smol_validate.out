-- Test smol_validate() function (line 2433)
-- This function is called by PostgreSQL during CREATE OPERATOR CLASS
-- It gets executed when the extension is created/loaded
CREATE EXTENSION IF NOT EXISTS smol;
NOTICE:  extension "smol" already exists, skipping
-- Query all smol operator classes (validates they were created successfully)
SELECT opcname, opfmethod.amname
FROM pg_opclass opc
JOIN pg_am opfmethod ON opc.opcmethod = opfmethod.oid
WHERE opfmethod.amname = 'smol'
ORDER BY opcname;
     opcname     | amname 
-----------------+--------
 bool_ops        | smol
 char_ops        | smol
 date_ops        | smol
 float4_ops      | smol
 float8_ops      | smol
 int2_ops        | smol
 int4_ops        | smol
 int8_ops        | smol
 interval_ops    | smol
 lsn_ops         | smol
 macaddr8_ops    | smol
 macaddr_ops     | smol
 money_ops       | smol
 name_ops        | smol
 oid_ops         | smol
 text_ops        | smol
 time_ops        | smol
 timestamp_ops   | smol
 timestamptz_ops | smol
 timetz_ops      | smol
 uuid_ops        | smol
(21 rows)

-- Verify that smol indexes can be created with all supported types
DROP TABLE IF EXISTS t_validate CASCADE;
NOTICE:  table "t_validate" does not exist, skipping
CREATE UNLOGGED TABLE t_validate (
    i2 int2,
    i4 int4,
    i8 int8,
    f4 float4,
    f8 float8,
    d date,
    t text
);
-- Create indexes to ensure operator classes are valid
CREATE INDEX idx_val_i2 ON t_validate USING smol(i2);
CREATE INDEX idx_val_i4 ON t_validate USING smol(i4);
CREATE INDEX idx_val_i8 ON t_validate USING smol(i8);
CREATE INDEX idx_val_f4 ON t_validate USING smol(f4);
CREATE INDEX idx_val_f8 ON t_validate USING smol(f8);
CREATE INDEX idx_val_d ON t_validate USING smol(d);
CREATE INDEX idx_val_t ON t_validate USING smol(t);
ERROR:  smol text keys require C/POSIX collation
-- Verify all indexes were created successfully
SELECT indexname, indexdef
FROM pg_indexes
WHERE tablename = 't_validate'
ORDER BY indexname;
 indexname  |                           indexdef                           
------------+--------------------------------------------------------------
 idx_val_d  | CREATE INDEX idx_val_d ON public.t_validate USING smol (d)
 idx_val_f4 | CREATE INDEX idx_val_f4 ON public.t_validate USING smol (f4)
 idx_val_f8 | CREATE INDEX idx_val_f8 ON public.t_validate USING smol (f8)
 idx_val_i2 | CREATE INDEX idx_val_i2 ON public.t_validate USING smol (i2)
 idx_val_i4 | CREATE INDEX idx_val_i4 ON public.t_validate USING smol (i4)
 idx_val_i8 | CREATE INDEX idx_val_i8 ON public.t_validate USING smol (i8)
(6 rows)

-- Cleanup
DROP TABLE t_validate CASCADE;
